React/todo/src/components/Item.jsx

```diff
@@ -5,37 +5,59 @@ function Item({ item, setTodos }) {
   const [editing, setEditing] = React.useState(false);
   const inputRef = React.useRef(null);
 
-  const toggleComplete = () => {
-    setTodos(prev =>
-      prev.map(t => (t.id === item.id ? { ...t, is_completed: !t.is_completed } : t))
-    );
-  };
+  function updateTodoById(id, updater) {
+    setTodos(prev => {
+      return prev.map(todo => {
+        if (todo.id !== id) return todo;
+        return updater(todo);
+      });
+    });
+  }
 
-  const deleteTodo = () => {
-    setTodos(prev => prev.filter(t => t.id !== item.id));
-  };
+  function toggleComplete() {
+    updateTodoById(item.id, todo => ({
+      ...todo,
+      is_completed: !todo.is_completed,
```

Comment:
Mixing different cases will confuse the future devs. Let's stick to one. In JS that is, CamelCase
```suggestion
      isCompleted: !todo.isCompleted,
```